backend/src/search/tests/search.controller.spec.ts

```diff
@@ -1,64 +1,110 @@
-    jest.spyOn(httpService['axiosRef'], 'get').mockImplementation((url: string) => {
-      const params = new URLSearchParams(new URL(url).search);
-      const keyword = params.get('query');
-      const rows = parseInt(params.get('rows'));
-      const selects = params.get('select').split(',');
-      const items = mockData.message.items.slice(0, rows).map((item, i) => {
-        return selects.reduce((acc, select) => {
-          if (select === 'title') item[select] = [`${keyword}-${i}`];
-          acc[select] = item[select];
-          return acc;
-        }, {});
-      });
-
-      return Promise.resolve<{ data: CrossRefResponse }>({
-        data: { message: { 'total-results': mockData.message['total-results'], items } },
-      });
-    });
+    const httpService = mockHttpService();
+    const rankingService = mockRankingService();
+    const elasticService = mockElasticService();
     const module: TestingModule = await Test.createTestingModule({
       controllers: [SearchController],
       providers: [
         SearchService,
+        {
+          provide: RankingService,
+          useValue: rankingService,
+        },
         {
           provide: HttpService,
           useValue: httpService,
         },
+        {
+          provide: ElasticsearchService,
+          useValue: elasticService,
+        },
       ],
     }).compile();
     controller = module.get(SearchController);
+    app = module.createNestApplication();
+    app.useGlobalPipes(new ValidationPipe());
+    await app.init();
   });
+  afterEach(() => app.close());
 
-  it('search controller 정의', () => {
-    expect(controller).toBeDefined();
-  });
+  describe('/search/auto-complete', () => {
+    it('getAutoCompletePapers - keyword=coffee 일 때 PaperInfo[]를 return', async () => {
+      // Case 1. elasticsearch에 data가 없을 경우
+      const keyword = 'coffee';
+      const itemsByCrossRef = await controller.getAutoCompletePapers({ keyword });
+      expect(itemsByCrossRef.length).toBe(5);
+      itemsByCrossRef.forEach((item) => {
+        expect(item instanceof PaperInfo).toBe(true);
+      });
 
-  it('getAutoCompletePapers - keyword=coffee 일 때 PaperInfo[]를 return', async () => {
-    const keyword = 'coffee';
-    const items = await controller.getAutoCompletePapers(keyword);
-    expect(items.length).toBe(5);
-    items.forEach((item) => {
-      expect(item).toBeInstanceOf(PaperInfo);
+      // Case 2. elasticsearch에 data가 있는 경우
+      const itemsByElasticsearch = await controller.getAutoCompletePapers({ keyword });
+      expect(itemsByElasticsearch.length).toBe(5);
+      itemsByElasticsearch.forEach((item) => {
+        expect(item instanceof PaperInfo).toBe(true);
+      });
+    });
+    it('keyword 미포함시 error - GET /search/auto-complete?keyword=', () => {
+      const url = (keyword: string) => `/search/auto-complete?keyword=${keyword}`;
+      return request(app.getHttpServer()).get(url('')).expect(400);
+    });
```

Comment:
`mockElasticService()` 함수에서 mockResolvedValueOnce로 elasticService.search의 목업 케이스 분리를 잘 해주셔서, 실제 테스트 코드내에서도 두 케이스가 다르게 동작하고 있는것을 확인했습니다.
그런데 현재의 코드는 두 케이스의 검증로직이 동일해서, 두 케이스가 실제로 구분되어 실행되고 있는지 확인하는 로직도 있으면 더 좋을 것 같아요.

```suggestion
      expect(spyGetElasticSearch).toBeCalledTimes(1);
      expect(spyGetCrossRefData).toBeCalledTimes(1);

      // Case 2. elasticsearch에 data가 있는 경우
      const itemsByElasticsearch = await controller.getAutoCompletePapers({ keyword });
      expect(itemsByElasticsearch.length).toBe(5);
      itemsByElasticsearch.forEach((item) => {
        expect(item instanceof PaperInfo).toBe(true);
      });

      expect(spyGetElasticSearch).toBeCalledTimes(2);
      expect(spyGetCrossRefData).toBeCalledTimes(1);
```
case1: spyGetElasticSearch, spyGetCrossRefData 모두 호출
case2: spyGetElasticSearch만 호출

이렇게 검증하면 되지 않을까요?